File: Easy/Two-Sum.py
def twoSum(self, nums, target):
  for i in nums:
      if target-i in nums:
        if nums.count(target-i) >= 2:
          indices = [index for index, x in enumerate(nums) if x == target-i]
          return [indices[0], indices[1]]
        elif nums.index(i) == nums.index(target-i): continue
        else: return [nums.index(i), nums.index(target-i)]

#Runtime: 804 ms, faster than 30.58% of Python3 online submissions for Two Sum.
#Memory Usage: 13.8 MB, less than 72.79% of Python3 online submissions for Two Sum.

File: Easy/Two-Sum/Two-Sum.js
// Runtime: 160 ms, faster than 18.00% of JavaScript online submissions for Two Sum.
// Memory Usage: 34.7 MB, less than 62.40% of JavaScript online submissions for Two Sum.
// 16/12/2019
// DD/MM/YY

function count(target, list) {
  let total = 0
  for (let i = 0; i < list.length; i++) {
    if (list[i] === target) {
      total ++
    }
  }
  return total
}

function indice(target, list) {
  let indices = [], i = -1
  while ((i = list.indexOf(target, i+1)) != -1){
      indices.push(i)
    }
  return indices
}

var twoSum = function(nums, target) {
  for (let x = 0; x < nums.length; x++) {
    let i = nums[x]
    if (nums.includes(target-i)) {
      if (count(target-i, nums) >= 2) {
        let indices = indice(target-i, nums)
        return [indices[0], indices[1]]
      } else if (nums.indexOf(i) === nums.indexOf(target-i)) {
        continue
      } else {
        return [nums.indexOf(i), nums.indexOf(target-i)]
      }
    }
  }
}